tests/test_compare_function.js

```diff
@@ -6,24 +6,31 @@ var queue = require('queue-async');
 
 if (process.argv.length !== 3) {
   console.log('\nUsage: node test_compare_function.js fixture_filename\n');
-  return;
-}
-
-var filename = process.argv[2];
-var jsonData = JSON.parse(fs.readFileSync(path.join('.', filename), 'utf-8'));
-var compareFunctionPath = path.join('../', 'comparators', jsonData.compareFunction);
-var compareFunction = require(compareFunctionPath);
+} else {
+  var filename = process.argv[2];
+  var jsonData = JSON.parse(fs.readFileSync(path.join('.', filename), 'utf-8'));
+  var compareFunctionPath = path.join(
+    '../',
+    'comparators',
+    jsonData.compareFunction
+  );
+  var compareFunction = require(compareFunctionPath);
 
-jsonData.fixtures.forEach(function (fixture) {
-  compareFunction(fixture.newVersion, fixture.oldVersion, function(error, result) {
-    console.log(fixture.description);
-    console.log('expected', fixture.expectedResult);
-    if (error) console.log(error);
-    console.log('actual', JSON.stringify(result), '\n');
-    if (JSON.stringify(fixture.expectedResult) !== JSON.stringify(result)) {
-      console.log('Test failed! Actual is not expected!');
-    } else {
-      console.log('Test passed! Actual is same as expected');
-    }
+  jsonData.fixtures.forEach(function(fixture) {
+    compareFunction(
+      fixture.newVersion,
+      fixture.oldVersion,
+      function(error, result) {
+        console.log(fixture.description);
+        console.log('expected', fixture.expectedResult);
+        if (error) console.log(error);
+        console.log('actual', JSON.stringify(result), '\n');
+        if (JSON.stringify(fixture.expectedResult) !== JSON.stringify(result)) {
+          console.log('Test failed! Actual is not expected!');
+        } else {
+          console.log('Test passed! Actual is same as expected');
+        }
+      }
+    );
   });
-});
+}
```

Comment:
The `yarn.lock` file is needed if you use [yarn](https://code.facebook.com/posts/1840075619545360). It locks all the dependencies from the `package.json`. It guarantees no dependencies error on a remote install.